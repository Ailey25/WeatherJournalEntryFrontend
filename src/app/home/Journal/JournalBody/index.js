import React from 'react';

export const JournalBody = (props) => {
  return(
    <div>
      <form onSubmit={(e) => props.handleSubmit(e)}>
        <section id="journal">
          <input id="title"
            placeholder="Enter journal title"
            value={props.title}
            onChange = {(e) => props.handleChange(e)}>
          </input>
        </section>
        <section>
          <textarea id="entry"
            rows="10" cols="50"
            placeholder="Your journal entry"
            value={props.entry}
            onChange={(e) => props.handleChange(e)}>
          </textarea>
        </section>
      </form>
    </div>
  )
};

export default JournalBody;
