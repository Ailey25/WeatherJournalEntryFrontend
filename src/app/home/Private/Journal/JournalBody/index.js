import React from 'react';

import { Input, TextArea } from '../styles';

export const JournalBody = (props) => {
  return(
    <form onSubmit={(e) => props.handleSubmit(e)}>
      <div className="column">
        <Input id="title"
          placeholder="Journal title"
          value={props.title}
          onChange = {(e) => props.handleChange(e)}
        />
        <TextArea id="entry"
          rows="10" cols="50"
          placeholder="Journal entry"
          value={props.entry}
          onChange={(e) => props.handleChange(e)}>
        </TextArea>
      </div>
    </form>
  )
};

export default JournalBody;
