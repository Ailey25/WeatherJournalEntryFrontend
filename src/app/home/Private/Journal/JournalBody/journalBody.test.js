import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount} from 'enzyme';

import { JournalBody } from '../JournalBody';

configure({ adapter: new Adapter() })

describe('JournalBody', () => {
  let props = {
    title: 'test title',
    entry: 'test entry'
  }

  it('renders correctly', () => {
    const wrapper = shallow(<JournalBody />);
    expect(wrapper).toMatchSnapshot();
  });

  it ('renders 1 title with correct value', () => {
    const wrapper = shallow(<JournalBody {...props}/>);
    const titleElement = wrapper.find('#title');
    expect(titleElement.length).toBe(1);
    expect(titleElement.props().value).toBe(props.title);
  });

  it ('renders 1 entry with correct value', () => {
    const wrapper = shallow(<JournalBody {...props}/>);
    const entryElement = wrapper.find('#entry');
    expect(entryElement.length).toBe(1);
    expect(entryElement.props().value).toBe(props.entry);
  });
});

describe('onChange', () => {
  describe('title', () => {
    let testProps = {};
    let journalBodyWrapper = {};

    beforeEach(() => {
      testProps = {
        handleChange: jest.fn(),
      }
      journalBodyWrapper = mount(
        <JournalBody handleChange={testProps.handleChange} />
      );
    });

    it('when there is onChange event', () => {
      const titleWrapperLength = journalBodyWrapper.find('#title').length;
      const titleWrapper = journalBodyWrapper.find('#title').at(titleWrapperLength - 1);
      const event = {
        currentTarget: {
          id: titleWrapper.props().id,
          value: "some title entered"
        }
      };
      expect(titleWrapper.simulate('change', event));
      expect(testProps.handleChange).toHaveBeenCalledTimes(1);
    });

    it('when there is no onChange event', () => {
      const titleWrapperLength = journalBodyWrapper.find('#title').length;
      const titleWrapper = journalBodyWrapper.find('#title').at(titleWrapperLength - 1);
      expect(testProps.handleChange).toHaveBeenCalledTimes(0);
    });
  });

  describe('entry', () => {
    let testProps = {};
    let journalBodyWrapper = {};

    beforeEach(() => {
      testProps = {
        handleChange: jest.fn(),
      }
      journalBodyWrapper = mount(
        <JournalBody handleChange={testProps.handleChange} />
      );
    });

    it('when there is onChange event', () => {
      const titleWrapperLength = journalBodyWrapper.find('#entry').length;
      const titleWrapper = journalBodyWrapper.find('#entry').at(titleWrapperLength - 1);
      const event = {
        currentTarget: {
          id: titleWrapper.props().id,
          value: "some entry entered"
        }
      };
      expect(titleWrapper.simulate('change', event));
      expect(testProps.handleChange).toHaveBeenCalledTimes(1);
    });

    it('when there is no onChange event', () => {
      const titleWrapperLength = journalBodyWrapper.find('#entry').length;
      const titleWrapper = journalBodyWrapper.find('#entry').at(titleWrapperLength - 1);
      expect(testProps.handleChange).toHaveBeenCalledTimes(0);
    });
  });
});
