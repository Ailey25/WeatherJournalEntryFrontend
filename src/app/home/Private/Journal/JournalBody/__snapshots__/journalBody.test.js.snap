// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalBody renders correctly 1`] = `
<div
  className="column"
>
  <Styled(styled.input)
    id="title"
    onChange={[Function]}
    placeholder="Journal title"
  />
  <Styled(styled.textarea)
    cols="50"
    id="entry"
    onChange={[Function]}
    placeholder="Journal entry"
    rows="10"
  />
</div>
`;
